### PR TITLE
Replace Rundeck with (Jenkins) pipelines Ticket: #283

### PR DESCRIPTION
**Pull request by kiwo**
Replace Rundeck with (Jenkins) pipelines. Adresses https://github.com/opendevstack/ods-project-quickstarters/issues/283.

### CODE REVIEW

File: jenkins/webhook-proxy/main.go [hunk 1]
```diff
@@ -326,12 +326,17 @@ func (s *Server) HandleRoot() http.HandlerFunc {
 		}
 
 		if event.Kind == "forward" {
-			gitURI := fmt.Sprintf(
-				"%s/%s/%s.git",
-				s.RepoBase,
-				project,
-				event.Repo,
-			)
+			var gitURI string;
```
Comment (michaelsauter) [inline jenkins/webhook-proxy/main.go]: Remove semicolon :)

File: create-projects/Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+def dockerRegistry
+
+
+node {
+  sharedLibraryRepository = env.SHARED_LIBRARY_REPOSITORY
+  dockerRegistry = env.DOCKER_REGISTRY
+}
+def conts = containerTemplate( 
+            name: 'jnlp',
+            image: "${dockerRegistry}/cd/jenkins-slave-base",
```
Comment (michaelsauter) [inline create-projects/Jenkinsfile]: Consider pointing to the next version: `"${dockerRegistry}/cd/jenkins-slave-base:2"`

File: create-projects/Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                doGenerateSubmoduleConfigurations: false, 
+                extensions: [[$class: 'RelativeTargetDirectory', relativeTargetDir: 'ods-config']],
+                submoduleCfg: [], 
+                userRemoteConfigs: [[credentialsId: 'prov-cd-cd-user-with-password', url: 'http://192.168.56.31:7990/scm/opendevstack/ods-configuration.git']] 
+            ])
+          }
+
+
+
+        stage('Creating Openshift Projects') {
```
Comment (michaelsauter) [inline create-projects/Jenkinsfile]: I'd use singular: `Creating Openshift Project` ... also please align upper/lowercase with the other stages

File: create-projects/create-projects.sh.bak [hunk 4]
```diff
@@ -0,0 +1,138 @@
+#!/bin/bash
```
Comment (michaelsauter) [inline create-projects/create-projects.sh.bak]: Should this file be removed?

File: infrastructure-setup/exports/master-realm.json [hunk 5]
```diff
@@ -0,0 +1,1402 @@
+{
```
Comment (michaelsauter) [inline infrastructure-setup/exports/master-realm.json]: Do we really want this file or should it rather be gitignored?

File: ocp-templates/ocp-config/component-environment/component-template.yml [hunk 6]
```diff
@@ -0,0 +1,146 @@
+apiVersion: v1
```
Comment (michaelsauter) [inline ocp-templates/ocp-config/component-environment/component-template.yml]: I would say that this folder should not be in `ods-core` - I would expect it somewhere in `ods-quickstarters/common`

File: ocp-templates/ocp-config/rshiny-app/rshiny-authproxy.yml [hunk 7]
```diff
@@ -0,0 +1,118 @@
+apiVersion: v1
```
Comment (michaelsauter) [inline ocp-templates/ocp-config/rshiny-app/rshiny-authproxy.yml]: Same for rshiny - remove from core!

File: create-projects/Jenkinsfile [hunk 8]
```diff
@@ -23,9 +17,7 @@ def conts = containerTemplate(
             args: ''
         )
 
-def pod_label = "hanswurst";
-
-
+def pod_label = "create-projects-" + projectId;
```
Comment (michaelsauter) [inline create-projects/Jenkinsfile]: Camel-case please ... could also be a random string right?

File: create-projects/Jenkinsfile (hunk 3 above)
Comment (kiwo) [reply create-projects/Jenkinsfile]: the stage will create multiple projects cd, dev, test.

Comment (michaelsauter) [reply create-projects/Jenkinsfile]: Ha, true!

**Comment by michaelsauter** [comment]
@kiwo Following up on our discussion earlier today:

* We definitely split the "create project" from the "create component" action - meaning you cannot do both at the same time. That frees you from checking that "create project" finished before you create the component inside it.
* The API interface to create projects MUST be stable. From my understanding, that is the case with your new implementation. Can you confirm this?
* The API interface to create components is ok to change

**Pull request merged**